icu4j/main/classes/core/src/com/ibm/icu/impl/number/LongNameMultiplexer.java

```diff
@@ -0,0 +1,104 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+
+package com.ibm.icu.impl.number;
+
+import com.ibm.icu.number.NumberFormatter;
+import com.ibm.icu.text.PluralRules;
+import com.ibm.icu.util.ICUException;
+import com.ibm.icu.util.MeasureUnit;
+import com.ibm.icu.util.NoUnit;
+import com.ibm.icu.util.ULocale;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LongNameMultiplexer implements MicroPropsGenerator {
+    private final MicroPropsGenerator fParent;
+
+    /**
+     * Because we only know which LongNameHandler we wish to call after calling
+     * earlier MicroPropsGenerators in the chain, LongNameMultiplexer keeps the
+     * parent link, while the LongNameHandlers are given no parents.
+     */
+    private List<LongNameHandler> fLongNameHandlers;
+    private List<MixedUnitLongNameHandler> fMixedUnitHandlers;
```

Comment:
I'm pretty sure you don't need these two fields in Java.  In C++ they are probably there because of memory management / ownership, but in Java you don't need to worry about that.

Reply:
you are right , done